addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+from openerp import models, fields, api, _
+
+class lunch_order(models.Model):
+    """
+    lunch order (contains one or more lunch order line(s))
+    """
+    _name = 'lunch.order'
+    _description = 'Lunch Order'
+    _order = 'date desc'
+
+    @api.multi
+    def name_get(self):
+        return [(order.id, _("Lunch Order #%d") % order.id) for order in self]
+
+    def _default_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        alert_msg = [alert.get_alert_message()
+                     for alert in self.env['lunch.alert'].search([])
+                     if alert.get_alert_message()]
+        return '\n'.join(alert_msg)
+
+    def _compute_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        if self.state == 'new':
+            self.alerts = self._default_alerts_get()
+
+    def _default_get_previous_order_ids(self):
+        prev_order = self.env['lunch.order.line'].search([('user_id', '=', self.env.uid)], limit=20 ,order='id desc')
+
+        # If we return return prev_order_res.ids, we will have duplicates (identical orders).
+        # Therefore, this following part removes duplicates based on product_id and note.
+
+        return {
+            (order.product_id, order.note): order.id
+            for order in prev_order
+        }.values()
+
+
+    @api.multi
+    def _compute_get_previous_order_ids(self):
+        for order in self:
+            order.previous_order_ids = order._default_get_previous_order_ids()
+
+    @api.multi
+    @api.depends('order_line_ids')
+    def _compute_total(self):
+        """
+        get and sum the order lines' price
+        """
+        for order in self:
+            order.total = sum(
+                orderline.price for orderline in order.order_line_ids)
+
+    @api.multi
+    def update_order_state(self):
+        """
+        Update the state of lunch.order based on its orderlines
+        """
+        for order in self:
+            isconfirmed = True
+            for orderline in order.order_line_ids:
+                if orderline.state == 'new':
+                    isconfirmed = False
+                if orderline.state == 'cancelled':
+                    isconfirmed = False
+                    order.state = 'partially'
+            if isconfirmed:
+                order.state = 'confirmed'
+
+
+    user_id = fields.Many2one('res.users', 'User Name', required=True, readonly=True,
+                              states={'new': [('readonly', False)]},
+                              default=lambda self: self.env.uid)
+
+    date = fields.Date('Date', required=True, readonly=True,
```

Comment:
I guess that if we do that, we lose the possibility to create orders for upcoming days.